frontend/src/components/common/button/button.tsx

```diff
@@ -1,17 +1,40 @@
+import { AppRoute } from 'common/enums/enums';
 import { FC } from 'common/types/types';
+import { getValidClasses } from 'helpers/helpers';
+import { useNavigate } from 'hooks/hooks';
 
 import styles from './styles.module.scss';
 
 type Props = {
   label: string;
+  btnColor?: 'blue' | 'gray';
   type?: 'button' | 'submit';
+  to?: AppRoute;
   onClick?: () => void;
 };
 
-const Button: FC<Props> = ({ type = 'button', label, onClick }) => (
-  <button type={type} className={styles.button} onClick={onClick}>
-    {label}
-  </button>
-);
+const Button: FC<Props> = ({
+  type = 'button',
+  label,
+  btnColor = 'blue',
+  to,
+  onClick,
+}) => {
+  const navigate = useNavigate();
+
+  const handleOnClick = (): void => {
+    to ? navigate(to) : onClick?.();
+  };
```

Comment:
```suggestion
  const isLink = Boolean(to)
  
  return isLink ? <AppLink> ? <Button>
```

Reply:
@what1s1ove for common component Link 'to' props is required but for the Button isn't, what default value has to be passed in such case?

Reply:
no one. since we add a condition
you can cast a type 

```
  const isLink = Boolean(to)
  
  return isLink ? <AppLink to={to as AppRoute}> ? <button></button>
```

Reply:
done